### ISSUE TITLE
V2.90 Recurring Events Timezone issue

### ISSUE THREAD

**Issue by HKpat** [phw198/OutlookGoogleCalendarSync#1265]
V2.90 is doing something weird with recurring appointments that was not happening in v2.8.4. (Outlook to Google)
It would appear to be to looking at the end time of the appoint and not the start time and then subtracting the UTC time difference away!!! I am in Hong Kong UTC+8 (both outlook and google set to this time zone)

Had to revert back to 2.8.4 as if you have say an appointment set up for 10 occurrences then it is deleting the 10th one from google because it's telling google to end the series the day before. The screenshot below is showing what happens when I do a full sync when reverting back to 2.84
![Screenshot 2021-08-23 123617](https://user-images.githubusercontent.com/89371946/130390619-0f6b0550-bfb7-4979-825a-15df7ff3fac7.png)

Differences in time
1st 6.5 hrs
2nd 7.25hrs
3rd 7.5 hrs

1st appointment is 1.5 hr long, 2nd 45 mins long, 3rd 30 mins long.
Adding the time differences and the appoint duration you always get 8hrs.

**Comment by phw198** [comment]
This will be related to #1035 and commit 2bcd32314c7fac709d7a6aa5ac5e4a0a40207e1e which was released in v2.8.5

Prompted by [this](https://stackoverflow.com/questions/53997143/valueerror-rrule-until-values-must-be-specified-in-utc-when-dtstart-is-timezone), do the Google calendar items for those affected items above show a timezone for the start date when you double click them to edit?

Would be useful if you could let OGCS recreate one of these appointments:
1. Set OGCS to O->G
2. Set logging level to `FINE`
3. Delete the Google appointment
4. Sync with OGCS to recreate it.
5. Provide the log lines related to the (re)creation of that appointment.

**Comment by HKpat** [comment]
_Prompted by this, do the Google calendar items for those affected items above show a timezone for the start date when you double click them to edit?_
The google calendar items show GMT+8 . 


Attached is the log file and screen shots for 2 recurring items as per the instructions. In both instances the google calendar does not show the final instance. If I revert back to 2.8.4 delete the google appoint and sync the final instances are all there.

[OCGS.txt](https://github.com/phw198/OutlookGoogleCalendarSync/files/7046663/OCGS.txt)

![OGCS1](https://user-images.githubusercontent.com/89371946/130785197-d406c004-b2c8-4bba-ab40-5b1221dff15e.png)


![OGCS2](https://user-images.githubusercontent.com/89371946/130785202-1f3885bf-68ce-4bc4-ab7e-cca9fe90ec6a.png)

**Comment by phw198** [comment]
OK, so confusingly your opening post gave evidence of recurring series that were _not_ all-day appointments - I don't believe there is actually a problem with the end date of these?

However, your screenshots of the Outlook recurrences above _are_ all day appointments, and I have now confirmed these are ending a day early due to converting the midnight end time from local time to UTC, which it shouldn't be doing.

:package: This should now be resolved with [v2.9.0.1.zip](https://github.com/phw198/OutlookGoogleCalendarSync/files/7071185/v2.9.0.1.zip) ([see instructions for applying](https://github.com/phw198/OutlookGoogleCalendarSync/wiki/Reporting-Problems#applying-a-hotfix)) - make sure you upgrade back to v.2.9.0 before applying the hotfix.

Please let me know how it goes!

**Comment by HKpat** [comment]
Apologies for the confusion.

I've given it a thorough test and this now working perfectly !!!

**Comment by HKpat** [comment]
Hi I just noticed something else related to the same issue. I have my "when" settings set to 365 days in the past and 365 in future, I also have confirm deletions set on at the moment and I noticed that the sync is finding 1 more entry in google than in outlook and trying to delete an entry from google from the 29th Aug 2020 (366 days ago). This is a recurring all day event also!

![Date](https://user-images.githubusercontent.com/89371946/131272103-9cfc57dd-a14c-4547-9a93-25412b365097.png)
![Delete](https://user-images.githubusercontent.com/89371946/131272116-dd468089-5c7d-42e3-ab23-5916d230eb5b.png)
